src/main/java/racingcar/service/RacingcarService.java

```diff
@@ -3,42 +3,43 @@
 import java.util.ArrayList;
 import java.util.List;
 import racingcar.domain.Car;
-import racingcar.dto.CarPerformance;
+import racingcar.domain.Cars;
 import racingcar.dto.RoundResult;
 import racingcar.util.RandomNumberGenerator;
 
 public class RacingcarService {
 
-    private static final int MIN_OIL_AMOUNT = 0;
-    private static final int MAX_OIL_AMOUNT = 9;
+    private static final int MIN_NUMBER_OF_POWER = 0;
+    private static final int MAX_NUMBER_OF_POWER = 9;
 
-    public RoundResult processRound(List<Car> cars) {
-        List<CarPerformance> carPerformances = new ArrayList<>();
-        for (Car car : cars) {
-            car.goIfOilEnough(decideOilAmount());
-            CarPerformance carPerformance = car.getCurrentPerformance();
-            carPerformances.add(carPerformance);
-        }
-        return new RoundResult(carPerformances);
+    public Cars createCars(List<String> carNames) {
+        List<Car> cars = carNames.stream()
+                .map(Car::new)
+                .toList();
+        return new Cars(cars);
     }
 
-    public List<String> pickOutWinners(List<Car> cars) {
-        int maxDistance = Car.findMaxDistance(cars);
-        List<Car> winners =  cars.stream()
-                .filter(car -> car.isSameDistance(maxDistance))
-                .toList();
-        return Car.mapCarsToName(winners);
+    public RoundResult processRound(Cars cars) {
+        int registeredCarsSize = cars.cars().size();
+        List<Integer> numbersOfPower = prepareNumbersOfPower(registeredCarsSize);
+        cars.raceWithNumbersOfPower(numbersOfPower);
+        return new RoundResult(cars);
     }
 
-    private int decideOilAmount() {
-        return RandomNumberGenerator.generate(MIN_OIL_AMOUNT, MAX_OIL_AMOUNT);
+    public List<String> pickOutWinners(Cars cars) {
+        int maxDistance = cars.findMaxDistance();
+        return cars.findCarNamesWithSameDistance(maxDistance);
     }
 
-    public List<Car> createCars(List<String> carNames) {
-        List<Car> cars = carNames.stream()
-                .map(Car::new)
-                .toList();
-        Car.validateUniqueCarNames(cars);
-        return cars;
+    private List<Integer> prepareNumbersOfPower(int registeredCarsSize) {
+        List<Integer> numbersOfPowers = new ArrayList<>();
+        for (int i=0; i<registeredCarsSize; i++) {
```

Comment:
코드 포맷팅을 해보면 띄어쓰기가 안된 곳이 있네용!
```suggestion
        for (int i = 0; i < registeredCarsSize; i++) {
```